src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -113,14 +114,13 @@ public String deleteProfile(String id, AuthenticatedUserRequest user) {
     public EnumerateBillingProfileModel enumerateProfiles(Integer offset,
                                                           Integer limit,
                                                           AuthenticatedUserRequest user) {
-        // TODO: add back once spend profile fully implemented
-        /*
-        List<UUID> resources = iamService.listAuthorizedResources(user, IamResourceType.SPEND_PROFILE);
-        if (resources.isEmpty()) {
-            return new EnumerateBillingProfileModel().total(0);
+        if (applicationConfiguration.isEnforceBillingProfileAuthorization()) {
+            List<UUID> resources = iamService.listAuthorizedResources(user, IamResourceType.SPEND_PROFILE);
+            if (resources.isEmpty()) {
+                return new EnumerateBillingProfileModel().total(0);
+            }
+            return profileDao.enumerateBillingProfiles(offset, limit, resources);
         }
```

Comment:
could you swap the if to be !resources.isEmpty() and only have two possible return statements instead of three? Aren't line 120 and 124 the same?

Reply:
I could, but this way, when I remove the temporary `isEnforceBillingProfileAuthorization` it is an easy edit.